pallets/pallet-bonded-coins/src/tests/transactions/refund_account.rs

```diff
@@ -266,6 +266,47 @@ fn refund_below_min_balance() {
 		});
 }
 
+#[test]
+fn refund_account_fails_when_account_blocked() {
```

Comment:
@abdulmth We have a test that checks the minimum balance case above this one, and I added a test here that ensures no balance is burnt when the beneficiary account cannot receive the funds due to any other reason (here: being blocked). This test fails if you revert the changes I made to the refund_account extrinsic.